plugin/src/main/kotlin/Mirakle.kt

```diff
@@ -207,6 +305,13 @@ class Mirakle : Plugin<Gradle> {
     }
 }
 
+class MirakleBreakMode : Mirakle() {
+    override fun apply(gradle: Gradle) {
+        gradle.startParameter.projectProperties = gradle.startParameter.projectProperties.plus(BREAK_MODE to "true")
```

Comment:
Could this just be done in the main plugin based on either a new flag, or `breakOnTasks` being non empty?

Reply:
No. The problem is that mirake needs to know whether to start or not in break mode before project evaluation and before that moment when `mirakle {}` extension will be available. 
If build launched in break mode, mirakle will let Gradle populate task graph and thus spend some time on local machine.
Configuring mirakle as soon as possible through `gradle.startParameter.projectProperties` will save that time for non-break builds.

Reply:
Makes sense. Thanks for the explanation.

Reply:
Maybe there is a better way to configure it, but I don't see it. Will be glad to hear your ideas.

Reply:
If I understand it correctly,

A good side effect of current implementation is that it should be possible to launch in BreakMode vs Non-BreakMode without modifying source code by using `apply plugin: Mirakle` and passing `-Pmirakle.break.mode=true/false` while invoking the gradle-build from command-line.

Reply:
Yup, and I realized that parameter value from CLI should have priority over MirakleBreakMode. Just made a fix for this.